src/test/run-make/native-link-modifier-whole-archive/c_static_lib_with_constructor.c

```diff
@@ -0,0 +1,13 @@
+#include <stdio.h>
+void __attribute__((constructor)) construct();
+void __attribute__((destructor)) destruct();
```

Comment:
`__attribute__((constructor))` and `__attribute__((destructor))` aren't supported by VC++. I'm not sure which compiler `cc` will use on the `-pc-windows-msvc` targets but we might need to skip this test if those targets use VC++ instead of clang.

Reply:
Good catch! I switched to a C++-based implementation that should have the same effect.